packages/vite/src/node/plugins/css.ts

```diff
@@ -2277,6 +2298,82 @@ const makeModernScssWorker = (
   return worker
 }
 
+// this is mostly a copy&paste of makeModernScssWorker
+// however sharing code between two is hard because
+// makeModernScssWorker above needs function inlined for worker.
+const makeModernCompilerScssWorker = (
+  resolvers: CSSAtImportResolvers,
+  alias: Alias[],
+  _maxWorkers: number | undefined,
+) => {
+  let compiler: Sass.AsyncCompiler | undefined
+
+  const worker: Awaited<ReturnType<typeof makeModernScssWorker>> = {
+    async run(sassPath, data, options) {
+      // need pathToFileURL for windows since import("D:...") fails
+      // https://github.com/nodejs/node/issues/31710
+      const sass: typeof Sass = (await import(pathToFileURL(sassPath).href))
+        .default
+      compiler ??= await sass.initAsyncCompiler()
+
+      const sassOptions = { ...options } as Sass.StringOptions<'async'>
+      sassOptions.url = pathToFileURL(options.filename)
+      sassOptions.sourceMap = options.enableSourcemap
+
+      const internalImporter: Sass.Importer<'async'> = {
+        async canonicalize(url, context) {
+          const importer = context.containingUrl
+            ? fileURLToPath(context.containingUrl)
+            : options.filename
+          const resolved = await resolvers.sass(url, cleanScssBugUrl(importer))
+          return resolved ? pathToFileURL(resolved) : null
+        },
+        async load(canonicalUrl) {
+          const ext = path.extname(canonicalUrl.pathname)
+          let syntax: Sass.Syntax = 'scss'
+          if (ext && ext.toLowerCase() === '.sass') {
+            syntax = 'indented'
+          } else if (ext && ext.toLowerCase() === '.css') {
```

Comment:
The toLowerCase can be removed like for modern

Reply:
Sorry for making you comment the same thing :sweat_smile: Fixed in https://github.com/vitejs/vite/pull/17754/commits/320b5a0c41e3ed41897ef3b3cd920b2c4fcbb761

This copy&paste is indeed not good, but trying to refactor these two would probably make it more complicated, so I'll keep that for now.

Reply:
No problem. The duplication is fine by me, always better to keep things simple!